src/Reflection/ReflectionClass.php

```diff
@@ -251,19 +262,54 @@ function (ClassMethod $methodNode) : ReflectionMethod {
                 $this->node->getMethods()
             ),
             ...array_map(
-                function (ReflectionClass $trait) : array {
-                    return array_map(function (ReflectionMethod $method) use ($trait) : ReflectionMethod {
-                        $methodAst = $method->getAst();
-                        assert($methodAst instanceof ClassMethod);
-
-                        return ReflectionMethod::createFromNode(
-                            $this->reflector,
-                            $methodAst,
-                            $this->declaringNamespace,
-                            $trait,
-                            $this
-                        );
-                    }, $trait->getMethods());
+                function (ReflectionClass $trait) use ($traitAliases, $traitPrecedences) : array {
+                    return array_merge(
+                        [],
+                        ...array_filter(
+                            array_map(
+                                function (ReflectionMethod $method) use ($traitAliases, $traitPrecedences) : ?array {
```

Comment:
this logic now contains looping and more complex checks: we need it to be split out into a `private` method